File: frontend/src/api/interface/group.ts
export namespace Group {
    export interface GroupInfo {
        id: number;
        name: string;
        type: string;
        isDefault: boolean;
        isDelete: boolean;
    }
    export interface GroupCreate {
        id: number;
        name: string;
        type: string;
    }
    export interface GroupUpdate {
        id: number;
        name: string;
        isDefault: boolean;
    }
}

File: agent/app/dto/group.go
package dto

type GroupCreate struct {
	ID   uint   `json:"id"`
	Name string `json:"name" validate:"required"`
	Type string `json:"type" validate:"required"`
}

type GroupSearch struct {
	Type string `json:"type" validate:"required"`
}

type GroupUpdate struct {
	ID        uint   `json:"id"`
	Name      string `json:"name"`
	Type      string `json:"type" validate:"required"`
	IsDefault bool   `json:"isDefault"`
}

type GroupInfo struct {
	ID        uint   `json:"id"`
	Name      string `json:"name"`
	Type      string `json:"type"`
	IsDefault bool   `json:"isDefault"`
}
